pkg/pagerduty/reciever/reciever.go

```diff
@@ -0,0 +1,55 @@
+package reciever
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+
+	sdk "github.com/PagerDuty/go-pagerduty"
+)
+
+// fileReader will wrap the os or fstest.MapFS stucts so we are not locked in
+type fileReader interface {
+	ReadFile(name string) ([]byte, error)
+}
+
+// ExtractExternalIDFromPayload will retrieve the payloadPath and return the externalID
+func ExtractExternalIDFromPayload(payloadFilePath string, reader fileReader) (string, error) {
+	data, err := readPayloadFile(payloadFilePath, reader)
+	if err != nil {
+		return "", fmt.Errorf("could not read the payloadFile: %w", err)
+	}
```

Comment:
and yes I would use the reader lib directly here, as you are not adding logic on top of it in the readPayloadFile.
So I would simply do something like.
```suggestion
	data, err := reader.ReadFile(payloadFilePath)
	if err != nil {
		return "", fmt.Errorf("could not read the payloadFile: %w", err)
	}
```

Reply:
i'll see what I can do

Reply:
I remember why that was.
we wanted to add comments to pull the payload from a file (which can change) so I extracted it to a function that can be converted/deleted once we have a different/better implementation

Reply:
not sure if I understand this right. But from what I understood, wouldn't you then rather define the function as follows?:
```

  func ExtractExternalIDFromPayload(payload []data){
           [..]
  }
  
  func ExtractExternalIDFromFile(filepath: string) {
    data, err := reader.ReadFile(payloadFilePath)
    if err != nil {
          return "", fmt.Errorf("could not read the payloadFile: %w", err)
     } 
    return ExtractExternalIDFromPayload(data)
  }
```
But anyway, may just be taste

Reply:
agreed this would be good, just didn't expose the API as I didn't see the usage of this yet, testing didn't require me a different API and we can extract this API later if we need it

Reply:
added a comment regarding this so I can address it when I need to